tests/reducer-test.js

```diff
@@ -588,6 +588,126 @@ describe('reducer', function () {
       var changedState = reducer(initialState, {type: actions.CHANGE_VALUE, value: true, bunsenId: 'foo.3'})
       expect(changedState.value).to.eql({foo: ['test', undefined, undefined, true]})
     })
+
+    describe('using autoClean', function () {
+      it('will not prune values when nested model prop has autoClean set to false', function () {
+        const model = {
+          type: 'object',
+          properties: {
+            things: {
+              type: 'array',
+              items: {
+                autoClean: false,
+                type: 'object',
+                properties: {
+                  name: {type: 'string'}
+                }
+              }
+            }
+          }
+        }
+
+        const initialState = {
+          errors: {},
+          validationResult: {warnings: [], errors: []},
+          value: {},
+          model: model
+        }
+
+        const value = {
+          things: [
+            {name: ''},
+            {name: ''}
+          ]
+        }
+
+        const changedState = reducer(initialState, {type: actions.CHANGE_VALUE, value: value, bunsenId: null})
+
+        expect(changedState.value).to.eql(value)
+      })
+
+      it('will not prune values when nested model prop has autoClean set to true', function () {
```

Comment:
Shouldn't this say `will prune` instead of `will not prune`

Reply:
yeah... *sad Charlie Brown music plays in background